### ISSUE TITLE
Bug:  Error while loading rule '@typescript-eslint/await-thenable'

### ISSUE THREAD

**Issue by ThePlenkov** [typescript-eslint/typescript-eslint#8924]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

Hi, I'd like to report a problem I'm facing and I would like to ask for a help to get it resolved.

So in my project ( example is here https://github.com/ThePlenkov/typescript-eslint-issue ) I would like to use in addition to typescript-eslint also another plugins, for example @nx/eslint-plugin.

As a result I have two files like this:
root  base file
```

import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';
import nxEslintPlugin from '@nx/eslint-plugin';

export default tseslint.config(
  eslint.configs.recommended,
  ...tseslint.configs.recommendedTypeChecked,
  {
    languageOptions: {
      parserOptions: {                
        EXPERIMENTAL_useProjectService: true,
      },
    },
  },
  { plugins: { '@nx': nxEslintPlugin } },
);
```
and library config
```import baseConfig from '../../eslint.config.js'
import tseslint from 'typescript-eslint';
import * as jsonParser from 'jsonc-eslint-parser';

export default tseslint.config(
  ...baseConfig,
  // {
  //   languageOptions: {
  //     parser: jsonParser,
  //   },
  //   files: ['**/*.json'],
  //   rules: { '@nx/dependency-checks': 'error' },
  // }
);
```

the problem is - if this block is commented - everything works perfectly. But in case if I want to use a custom parser, let's say jsonParser for json files  like this:
```
  {
    languageOptions: {
      parser: jsonParser,
    },
    files: ['**/*.json'],
    rules: { '@nx/dependency-checks': 'error' },
  }
```

eslint fails with a following error:
```
eslint .


Oops! Something went wrong! :(

ESLint: 8.57.0

Error: Error while loading rule '@typescript-eslint/await-thenable': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
Parser: undefined
Note: detected a parser other than @typescript-eslint/parser. Make sure the parser is configured to forward "parserOptions.project" to @typescript-eslint/parser.
Occurred while linting /workspaces/typescript-eslint-issue/libs/a/package.json
    at throwError (/workspaces/typescript-eslint-issue/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:40:11)
    at getParserServices (/workspaces/typescript-eslint-issue/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:20:9)
    at create (/workspaces/typescript-eslint-issue/node_modules/@typescript-eslint/eslint-plugin/dist/rules/await-thenable.js:46:55)
    at Object.create (/workspaces/typescript-eslint-issue/node_modules/@typescript-eslint/utils/dist/eslint-utils/RuleCreator.js:38:20)
    at createRuleListeners (/workspaces/typescript-eslint-issue/node_modules/eslint/lib/linter/linter.js:895:21)
    at /workspaces/typescript-eslint-issue/node_modules/eslint/lib/linter/linter.js:1066:110
    at Array.forEach (<anonymous>)
    at runRules (/workspaces/typescript-eslint-issue/node_modules/eslint/lib/linter/linter.js:1003:34)
    at Linter._verifyWithFlatConfigArrayAndWithoutProcessors (/workspaces/typescript-eslint-issue/node_modules/eslint/lib/linter/linter.js:1730:31)
    at Linter._verifyWithFlatConfigArray (/workspaces/typescript-eslint-issue/node_modules/eslint/lib/linter/linter.js:1861:21)
Warning: command "eslint ." exited with non-zero status code
```

Do you recognize such a behavior? Do you think it is possible to adjust config to use let's say a custom parser only for *.json files? Thanks!

### Reproduction Repository Link

https://github.com/ThePlenkov/typescript-eslint-issue

### Repro Steps

1) clone the repo
2) `npm install`
3) `npx nx lint a` - it should work, lint will fail but with a real check
4) `npx nx lint b` - it should not work, lint will fail with mentioned above issue

### Versions

npx eslint -v
v8.57.0

**Comment by bradzacher** [comment]
This FAQ article should help answer your question.
https://typescript-eslint.io/troubleshooting/#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file

TL;DR type-aware rules don't magically get disabled just because you switch parsers. If you turn them on and don't provide types - they fail. You need to either provide types or turn them off.
Cos this is a json file you're linting - the latter is the correct course of action.
